tests/unit/hive_metastore_client/test_hive_metastore_client.py

```diff
@@ -434,3 +434,45 @@ def test_get_partition_keys_names_with_partitioned_table(
         mocked_get_partition_keys_objects.assert_called_once_with(
             db_name=database_name, table_name=table_name
         )
+
+    @mock.patch.object(HiveMetastoreClient, "drop_partition", return_value=None)
+    def test_bulk_drop_partitions(self, mock_drop_partition, hive_metastore_client):
+        # arrange
+        db_name = "db_name"
+        table_name = "table_name"
+        partition_list = [["1995", "9", "22"], ["2013", "2", "14"], ["2021", "1", "1"]]
+
+        # act
+        hive_metastore_client.bulk_drop_partitions(
+            db_name, table_name, partition_list, mock.ANY
+        )
+
+        # assert
+        assert mock_drop_partition.call_count == len(partition_list)
+
+    @mock.patch.object(HiveMetastoreClient, "drop_partition", return_value=None)
+    def test_bulk_drop_partitions_with_errors(
+        self, mock_drop_partition, hive_metastore_client
+    ):
+        # arrange
+        db_name = "db_name"
+        table_name = "table_name"
+        partition_list = [["1995", "9", "22"], ["2021", "1", "1"], ["2021", "1", "2"]]
+        mock_drop_partition.side_effect = [
+            None,
+            NoSuchObjectException(),
+            NoSuchObjectException(),
+        ]
+
+        # assert
+        with raises(NoSuchObjectException) as e:
```

Comment:
What about using `assertRaisesRegex`? Instead of validating the whole message we can validate only the part that makes sense testing.

Reply:
I did with the pytest.raises matches argument =].
Tks!